db-developments/python/aggregate.py

```diff
@@ -0,0 +1,105 @@
+        "additional_columns": [
+            ("cdtaname", "cdtaname{{decade[:-2]}}")
+        ],
+        "join_table": "dcp_cdta{{decade}}"
+    },
+    "commntydst": {
+        "source_column": "comunitydist",
+        "join_table": "dcp_cdboundaries",
+        "right_join_column": "borocd::TEXT"
+    },
+    "councildst": {
+        "source_column": "councildist",
+        "join_table": "dcp_councildistricts",
+        "right_join_column": "coundist::TEXT"
+    },
+    "nta": {
+        "source_column": "nta{{decade}}",
+        "output_column": "nta{{decade}}",
+        "additional_columns": [
+            ("ntaname", "ntaname{{decade[:-2]}}")
+        ],
+        "join_table": "dcp_nta{{decade}}"
+    },
+    "tract": {
+        "source_column": "bct{{decade}}",
+        "output_column": "bct{{decade}}",
+        "additional_columns": [
+            ("geoid", "centract{{decade}}")
+        ],
+        "group_by": ["boro", "bct{{decade}}", "centract{{decade}}"],
+        "join_table": "dcp_ct{{decade}}",
+        "right_join_column": "boroct2020"
+    }
+}
+
+if __name__ == "__main__":
+    filename = sys.argv[1]
+    decade = sys.argv[2]  # e.g. "10" or "20"
+    CAPTURE_DATE = os.environ.get("CAPTURE_DATE")
+
+    # Get current year
+    current_year = datetime.today().year
+
+    # SQL Template
+    with open(filename, 'r') as f:
+        sql = f.read()
+
+    try:
+        geom=sys.argv[3]
+    except:
+        print(f"Aggregating by year for decade {decade}")
+        sql_rendered = Template(sql).render(
+            years=list(range(2010, current_year+1)),
+            decade=decade,
+            CAPTURE_DATE=CAPTURE_DATE)
+    else:
+        print(f"Aggregating by {geom}")
+        column_info = geoms[geom]
+        source_column = column_info.get("source_column", geom)
+        output_column = column_info.get("output_column", geom)
+        output_column_internal = column_info.get("output_column_internal", output_column)
+        additional_columns = column_info.get("additional_columns", [])
+        group_by = column_info.get("group_by", [output_column])
+        join_table = column_info["join_table"]
+        right_join_column = column_info.get("right_join_column", output_column)
+
+        # Render template in twice because decade is part of render inputs as well
+        sql_rendered = Template(sql).render(
+            years=list(range(2010, current_year+1)),
+            CAPTURE_DATE=CAPTURE_DATE,
+            decade=decade,
+            geom=geom,
+            source_column=source_column,
+            output_column=output_column,
+            output_column_internal=output_column_internal,
+            additional_columns=additional_columns,
+            group_by=group_by,
+            join_table=join_table,
+            right_join_column=right_join_column)
+        
+        sql_rendered = Template(sql_rendered).render(decade=decade)
+    
+    with engine.begin() as connection:
+        connection.execute(text(sql_rendered))
```

Comment:
renamed from "yearly.py" with a bunch of changes. All the geom sql files were very repetitive so tried to abstract out main logic. This might be a tad sloppy overall (or more hard to follow - don't know if I love just having all of this in dictionaries). Would like your thoughts on this file/approach overall

Reply:
definitely a nice approach to abstract stuff to a single function. I could image a namedtuple _maybe_ being nicer than a dictionary, but probably not worth the effort in this PR

Reply:
I hope/think things will become a little nicer doing this with dbt instead of just using jinja with python 🤞